From 8ecd7c75b5a61498f0ff96110731e7671cc922da Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 08 Dec 2021 12:07:17 +0200
Subject: [PATCH] [SW-63186] gaudi2: security: set axprot for edma

EDMA is configured as user level access using axprot registers.

Change-Id: Id8190da8c81b50401fe6d46db4f86c5e7664d059
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index b1703be..457e16a 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -48,6 +48,8 @@
 #include <dcore0_mme_wb0_mstr_if_core_hbw_regs.h>
 #include <dcore0_mme_ctrl_mstr_if_core_hbw_regs.h>
 #include <dcore0_mme_ctrl_mstr_if_core_lbw_regs.h>
+#include <dcore0_edma0_mstr_if_core_hbw_regs.h>
+#include <dcore0_edma0_mstr_if_core_lbw_regs.h>
 
 /*
  * instances_num : Number of instances of RR to program.
@@ -213,6 +215,7 @@
 	AXPROT_HMMU,
 	AXPROT_TPC,
 	AXPROT_MME,
+	AXPROT_EDMA,
 };
 
 #define DCORE_DIST	0x200000
@@ -952,7 +955,10 @@
 #define MME_INST_NUM		NUM_OF_DCORES
 #define MME_INST_DIST		DCORE_DIST
 #define MME_CRTL_RTR_INST_NUM	1
-
+#define EDMA_RTR_INST_NUM	2
+#define EDMA_RTR_INST_DIST	0x10000
+#define EDMA_INST_NUM		NUM_OF_DCORES
+#define EDMA_INST_DIST		DCORE_DIST
 
 static struct axprot_regs axprot_reg[] = {
 	[AXPROT_VDEC] = {
@@ -994,6 +1000,14 @@
 		.dcore_num = MME_INST_NUM,
 		.inst_dist = MME_SBTE_RTR_INST_DIST,
 		.inst_num = MME_SBTE_RTR_INST_NUM,
+	},
+	[AXPROT_EDMA] = {
+		.base_reg = mmDCORE0_EDMA0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
+		.value = AXPROT_USER_VAL,
+		.dcore_dist = EDMA_INST_DIST,
+		.dcore_num = EDMA_INST_NUM,
+		.inst_dist = EDMA_RTR_INST_DIST,
+		.inst_num = EDMA_RTR_INST_NUM,
 	},
 };
 
@@ -1108,6 +1122,18 @@
 	preboot_axprot_regs_init(AXPROT_MME);
 }
 
+static void preboot_secure_edma(void)
+{
+	/* Configure edma to user level */
+	/* HBW AXPROT */
+	preboot_axprot_regs_init(AXPROT_EDMA);
+
+	axprot_reg[AXPROT_EDMA].base_reg = mmDCORE0_EDMA0_MSTR_IF_CORE_LBW_AXPROT_OVRD;
+
+	/* LBW AXPROT */
+	preboot_axprot_regs_init(AXPROT_EDMA);
+}
+
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -1124,6 +1150,7 @@
 		preboot_secure_trace();
 		preboot_secure_tpc();
 		preboot_secure_mme();
+		preboot_secure_edma();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
